js/src/Ice/Stream.d.ts

```diff
@@ -135,64 +135,293 @@ declare module "ice" {
+             *
+             * @param v The integer to write to the stream.
+             */
             writeInt(v: number): void;
+
+            /**
+             * Writes a long to the stream.
+             *
+             * @param v The long to write to the stream.
+             */
             writeLong(v: Ice.Long): void;
+
+            /**
+             * Writes a float to the stream.
+             *
+             * @param v The float to write to the stream.
+             */
             writeFloat(v: number): void;
+
+            /**
+             * Writes a double to the stream.
+             *
+             * @param v The double to write to the stream.
+             */
             writeDouble(v: number): void;
+
+            /**
+             * Writes a string to the stream.
+             *
+             * @param v The string to write to the stream. Passing null causes an empty string to be written to the
+             * stream.
+             */
             writeString(v: string): void;
+
+            /**
+             * Writes a proxy to the stream.
+             *
+             * @param v The proxy to write.
+             */
             writeProxy(v: ObjectPrx): void;
+
+            /**
+             * Writes an optional proxy to the stream.
+             *
+             * @param tag The optional tag.
+             * @param v The proxy to write.
+             */
             writeOptionalProxy(tag: number, v?: ObjectPrx | undefined): void;
 
             /// TODO use a base enum type
+            /**
+             * Writes an enumerated value.
+             * @param v The enumerator.
+             */
             writeEnum(v: any): void;
 
+            /**
+             * Writes a class instance to the stream. This method writes the index of an instance; the state of the
+             * value is written once {@link OutputStream.writePendingValues} is called.
+             *
+             * @param v The value to write.
+             */
             writeValue(v: Ice.Value): void;
+
+            /**
+             * Writes a user exception to the stream.
+             *
+             * @param v The user exception to write.</param>
+             */
             writeException(e: UserException): void;
-            writeUserException(e: UserException): void;
 
+            /**
+             *  Returns whether the stream is empty.
+             *
+             * @returns True if no data has been written yet, false otherwise.</returns>
+             */
             isEmpty(): boolean;
 
+            /**
+             * Expand the stream to accept more data.
+             *
+             * @param n The number of bytes to accommodate in the stream.
+             */
             expand(n: number): void;
 
-            //
-            // Sets the encoding format for class and exception instances.
-            //
+            /**
+             * Gets or sets the encoding format for class and exception instances.
```

Comment:
Are you sure this is needed? _format is read-only and private in other language mappings.

Reply:
Removed.